controller/src/main/java/org/jboss/as/controller/AttributeParsers.java

```diff
@@ -239,13 +239,21 @@ public void parseElement(AttributeDefinition attribute, XMLExtendedStreamReader
         static void parseEmbeddedElement(ObjectTypeAttributeDefinition attribute, XMLExtendedStreamReader reader, ModelNode op, String... additionalExpectedAttributes) throws XMLStreamException {
             AttributeDefinition[] valueTypes = attribute.getValueTypes();
 
-            Map<String, AttributeDefinition> attributes = Arrays.stream(valueTypes)
-                    .collect(Collectors.toMap(AttributeDefinition::getXmlName, Function.identity()));
-
-            Map<String, AttributeDefinition> attributeElements = Arrays.stream(valueTypes)
-                                   .filter(attributeDefinition -> attributeDefinition.getParser().isParseAsElement())
-                    .collect(Collectors.toMap(a -> a.getParser().getXmlName(a) , Function.identity()));
+            Map<String, AttributeDefinition> attributes = new HashMap<>();
+            for (AttributeDefinition valueType : valueTypes) {
+                if (attributes.put(valueType.getXmlName(), valueType) != null) {
+                    throw new IllegalStateException("Duplicate key");
```

Comment:
This technically should be i18n, the overhead of which probably defeats the purpose of this PR. But really the duplicate check should just be removed. It was a side effect of Collectors.toMap and doing that kind of check is not the responsibility of this method. So we can save more code. ;)